event/network.go

```diff
@@ -0,0 +1,13 @@
+package event
+
+import "github.com/libp2p/go-libp2p-core/network"
+
+// EvtPeerStateChange should be emitted everytime we form a connection with a peer or drop our last
+// connection with the peer. Essentially, it is emitted in two cases:
+// a) We form a/any connection with a peer.
+// b) We go from having a connection/s with a peer to having no connection with the peer.
+// It contains the connection handle and the new connection state.
+type EvtPeerStateChange struct {
+	Connection network.Conn
```

Comment:
I'm not actually sure if the connection should be a part of this event. It should probably just be the peer ID as there could be multiple connections (the service can query for those connections if necessary).

Reply:
Agree; and this could prevent the connection from being GC'ed if the event is queued for too long and we disconnect.